### ISSUE TITLE
azure-core: no module named cchardet

### ISSUE THREAD

**Issue by skshetry** [Azure/azure-sdk-for-python#21516]
**Describe the bug**

It looks like azure-core depends on a dependency that aiohttp brings with `chardet`, which has been replaced with [`charset-normalizer`](https://github.com/aio-libs/aiohttp/blob/dccf502fbcb4bd0f5ab1c7dd5ae1fe69541f4f39/aiohttp/client_reqrep.py#L75). So, now chardet is unavailable but azure-core still looks for the [`chardet`](https://github.com/Azure/azure-sdk-for-python/blob/9f110dd8b2c565c493aca135f9fc5b1c6e942167/sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py#L32). This transition from `chardet` to `chardet-normalizer` happened in the newest release of aiohttp v3.8.0: https://github.com/aio-libs/aiohttp/releases/tag/v3.8.0.

***Exception or Stack Trace***

```python
	  import sys
	  from typing import (
	      Any, Optional, AsyncIterator as AsyncIteratorType, TYPE_CHECKING, overload
	  )
	  from collections.abc import AsyncIterator
	  try:
>       import cchardet as chardet
E       ModuleNotFoundError: No module named 'cchardet'

/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/azure/core/pipeline/transport/_aiohttp.py:32: ModuleNotFoundError

During handling of the above exception, another exception occurred:

name = 'AioHttpTransport'

    def __getattr__(name):
        if name == 'AioHttpTransport':
            try:
>               from ._aiohttp import AioHttpTransport

/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/azure/core/pipeline/transport/__init__.py:68: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    import sys
    from typing import (
        Any, Optional, AsyncIterator as AsyncIteratorType, TYPE_CHECKING, overload
    )
    from collections.abc import AsyncIterator
    try:
        import cchardet as chardet
    except ImportError:  # pragma: no cover
>       import chardet  # type: ignore
E       ModuleNotFoundError: No module named 'chardet'

/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/azure/core/pipeline/transport/_aiohttp.py:34: ModuleNotFoundError

During handling of the above exception, another exception occurred:

    def test_azure_login_methods():
        def get_login_method(config):
            fs = AzureFileSystem(**config)
            # pylint: disable=pointless-statement
            return fs.login_method
    
        with pytest.raises(AzureAuthError):
            get_login_method(***)
    
        assert (
            get_login_method(***"connection_string": "test"***) == "connection string"
        )
>       assert get_login_method(***"account_name": "test"***).startswith(
            "default credentials"
        )

tests/unit/fs/test_azure.py:62: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/unit/fs/test_azure.py:52: in get_login_method
    fs = AzureFileSystem(**config)
dvc/fs/fsspec_wrapper.py:44: in __init__
    self.fs_args.update(self._prepare_credentials(**kwargs))
dvc/fs/azure.py:122: in _prepare_credentials
    "exclude_managed_identity_credential", False
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/azure/identity/aio/_credentials/default.py:108: in __init__
    credentials.append(ManagedIdentityCredential(client_id=managed_identity_client_id, **kwargs))
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/azure/identity/aio/_credentials/managed_identity.py:83: in __init__
    self._credential = ImdsCredential(**kwargs)
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/azure/identity/aio/_credentials/imds.py:26: in __init__
    self._client = AsyncManagedIdentityClient(get_request, **dict(PIPELINE_SETTINGS, **kwargs))
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/azure/identity/_internal/managed_identity_client.py:42: in __init__
    self._pipeline = self._build_pipeline(**kwargs)
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/azure/identity/aio/_internal/managed_identity_client.py:39: in _build_pipeline
    return build_async_pipeline(**kwargs)
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/azure/identity/_internal/pipeline.py:80: in build_async_pipeline
    from azure.core.pipeline.transport import AioHttpTransport
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

name = 'AioHttpTransport'

    def __getattr__(name):
        if name == 'AioHttpTransport':
            try:
                from ._aiohttp import AioHttpTransport
                return AioHttpTransport
            except ImportError:
>               raise ImportError("aiohttp package is not installed")
E               ImportError: aiohttp package is not installed

/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/azure/core/pipeline/transport/__init__.py:71: ImportError
```

**To Reproduce**
1. Install `azure-core` and `aiohttp` together.

**Comment by james-ki** [comment]
Pinning `aiohttp = "<3.8.0"`, as mentioned in [the link above](https://github.com/iterative/dvc/issues/6899) has worked as a temporary fix for me

**Comment by xiangyan99** [comment]
Thanks for reporting the issue. 

aiohttp made a change to not install chardet and unfortunately it brake azure-core.

We released a patch https://pypi.org/project/azure-core/1.19.1/ to address it.

Please let us know if it does not work.

Thank you.

**Comment by skshetry** [comment]
Thanks, I can confirm that it's fixed in 1.19.1.
